aiokatcp/sensor.py

```diff
@@ -75,47 +106,63 @@ def __init__(self, sensor_type: Type[_T],
                  description: str = None,
                  units: str = '',
                  default: _T = None,
-                 initial_status: Status = Status.UNKNOWN) -> None:
+                 initial_status: Status = Status.UNKNOWN,
+                 *,
+                 status_func: Callable[[_T], Status] = _default_status_func) -> None:
         self.stype = sensor_type
         type_info = core.get_type(sensor_type)
         self.type_name = type_info.name
         self._observers = set()           # type: Set[Callable[[Sensor[_T], Reading[_T]], None]]
         self.name = name
         self.description = description
         self.units = units
+        self.status_func = status_func
```

Comment:
Relatively surprised you allow status_func to be changed after creation. Is there a solid use case for this, otherwise it might be preferable to discourage it.

Reply:
More a case of there being no particular reason to disallow it, and all the other attributes are public so it felt odd to make this one private.

One use case is that some thresholds may only be known later e.g. we generally expect to get two incomplete heaps per X engine (one at the start, one at the end) and might only want to warn if there are more than this, but the number of X engines might only be known later.